src/golang/lib/job/config.go

```diff
@@ -64,6 +64,8 @@ type DatabricksJobManagerConfig struct {
 	// storage in S3. Information on how to create this can be found here:
 	// https://docs.databricks.com/aws/iam/instance-profile-tutorial.html
 	S3InstanceProfileARN string `yaml:"s3InstanceProfileArn" json:"s3_instance_profile_arn"`
+	// [Optional] ID of instance pool that Aqueduct-created JobClusters should use.
+	InstancePoolID *string `yaml:"instancePoolID" json:"instance_pool_id"`
```

Comment:
Any specific reason for this to be a pointer?

Reply:
It's an optional value, I've read somewhere that using pointers for optional values is better than just dealing with empty strings.